Backend/src/main/java/idealab/api/operations/PrintJobOperations.java

```diff
@@ -76,19 +78,37 @@ public GenericResponse deletePrintJobStatus(PrintJobDeleteRequest dto) {
     }
 
     public GetAllPrintJobListResponse getAllPrintJobs(){
-        // Some magic happened here !
-        //TODO: Repo class should be coded !
-        //TODO: They should be connected !
-        //TODO: Domain entity (model) should be mapped response. It will be better if we use dto then map to response to make it more flexible for future use.
-        //TODO: but now we don't need dto as mid entity between response entity and database entity.
+        Iterable<PrintJob> allJobs = printJobRepo.findAll();
+        List<GetAllPrintJobResponse> printJobResponses = new ArrayList<GetAllPrintJobResponse>();
+
+        allJobs.forEach(printJob -> {
```

Comment:
I would imagine SpringBoot/Jackson has a way of doing this serialization for us but I'm not sure. At the very least can you make a constructor method of GetAllPrintJobResponse (renamed to PrintJobResponse) that takes in a PrintJob model and does the conversion for us? That way we don't have to have this type of conversion code littered around our code base each time we want to convert fromPrintJob model -> PrintJobResponse.

Reply:
@e-carlin I think I added something like this when I was working on my Dropbox branch.  Check out https://github.com/codefordenver/ideaLab/blob/master/Backend/src/main/java/idealab/api/dto/response/GetPrintJobDataResponse.java

Reply:
Yep, thanks! I think it could be renamed as well. Something like PrintJobListResponse to imply that it has a list and the list is of printjobs. I also find it annoying that we have to do all of this hand translation between objects that share almost the exact same structure. There must be a more clever way of mapping between the two I'm just not sure what it is. Having a response, request, and model object that all share very identical structures feels like overkill.